src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -83,6 +83,7 @@ public static void ToCharsBuffer(byte value, Span<char> buffer, int startingInde
             buffer[startingIndex] = (char)(packedResult >> 8);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Was this not getting inlined  already? I wonder why

Reply:
Methods with loops are generally too large to inline by default.

To confirm, you can try:
* BDN's [inlining diagnoser](https://benchmarkdotnet.org/articles/samples/IntroInliningDiagnoser.html)
* enable and look at the jit inlining events under perfview
* Use a checked jit and set `COMPlus_JitPrintInlinedMethods=<class_name>.<method name>`

Reply:
> Was this not getting inlined already? I wonder why

It is considered unprofitable... (as per BDN inlining diagnoser)

```
--------------------
Inliner: System.HexConverter+<>c.<ToString>b__4_0 - instance void  (value class System.Span`1<wchar>,value class System.ValueTuple`3<int,int32,value class Casing>)
Inlinee: System.HexConverter.EncodeToUtf16 - void  (value class System.ReadOnlySpan`1<unsigned int8>,value class System.Span`1<wchar>,value class Casing)
Fail Reason: unprofitable inline
--------------------
```